opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,36 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (bool, str, Number, Sequence)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, Sequence) and len(value) > 0:
```

Comment:
We can use `elif` here to avoid checking `isinstance(value, Sequence)` twice.

Reply:
Since there is a return statement in the the first `if`, this seems to be the equivalent of an `if/elif` statement unless I am misunderstanding something. Is there a best practice or some kind of optimization here that I'm unfamiliar with that suggests we should add an `elif` here?

Reply:
Sorry, that was a quite vague of me. What I meant was that we could rearange the checks like this:
```python
if  isinstance(value, Sequence):
   if len(value) > 0:
      # ...
elif not not isinstance(value, (bool, str, Number)):
   # ...
```
to avoid checking for `isinstance(value, Sequence)` twice in the case where the first `if not isinstance`-branch is not entered.

Reply:
Ah that makes sense. Good suggestion.

Reply:
For empty sequences, we still check `isinstance(value, Sequence)` twice now. We could either break the `and len(value) > 0` into a new `if` like I suggested above, or have `_check_attribute_value_sequence` handle empty sequences (by just returning `None`) and remove the `and len` condition.

Reply:
Now that you point it out, the `len()` check definitely belongs inside the validation method.